### ISSUE TITLE
Make pyenchant an optional dependency

### ISSUE THREAD

**Issue by codingjoe** [flashcode/msgcheck#10]
Since [pyenchant](https://github.com/rfk/pyenchant) is no longer maintained, I would recommend to make it an optional dependency via `extras_require`.
That would mean you would need to run `pip install msgcheck[pyenchant]` or `pip install msgcheck[spelling]` to install pyenchant and spell checking support.

What do you think @flashcode ? Currently I can't install this package on travis-ci because of the hard pyenchant dependency.

**Comment by flashcode** [comment]
Hi @codingjoe,
Maybe a better long-term solution would be to replace pyenchant by something else that is activel maintained?

**Comment by codingjoe** [comment]
@flashcode yes, that would be cool, but I don't see anything on the horizon. The only thing open source tool, that provides proper spell-checking is languagetools. Aspell or hundspell are all pretty mediocre. I think a local spelling dictionary can never get close to an online crowd sourced one.

I would be happy to implement languagetools support, but people would need to set up their own server or provide an API token.

Anyways, I don't know if it isn't a good move to make spell checking optional anyways. No matter how it's implemented, since it will never work without a 3rd party dependency. Something like `pip install msgcheck[spelling]` would be future prove.

**Comment by codingjoe** [comment]
@flashcode I did some digging. I would recommend to use [pyLanguagetool](https://github.com/Findus23/pyLanguagetool). This seems to be a pretty good. What do you think?
